### ISSUE TITLE
The number of causals when running MCMC

### ISSUE THREAD

**Issue by variani** [gkichaev/PAINTOR_V3.0#23]
Hi,

I am trying to run MCMC (`-mcmc` argument) and specify the number of causal snps (`-max_causal` argument). It seems that the MCMC procedure doesn't care about `-max_causal`. Do you have any specification on the number of causal snps and their prior probabilities when running MCMC?

When running `-enumerate 2`, the output seems consistent. That is, the sum of posterior probabilities of snps to be causal sums up to 2. However, this sum value is something random, e.g. 4.7 or 6.4, when running `-mcmc` instead.

Basically, I would expect the same behavior as in FINEMAP (http://www.christianbenner.com/), where the controls are given with `--n-causal-snps` and `--prior-k` arguments.

Thanks.

**Comment by variani** [comment]
Thanks for your helpful comments.

One of my motivation to address this parametrization was exactly comparisons, e.g. FINEMAP and PAINTOR. But it seems to me not possible without control of  similar `--n-causal-snps` and `--prior-k` arguments in PAINTOR (running practical examples gives the same conclusions). Further, I don't see the way to compare PAINTOR with its two runs, `-enumerate 3` and `-mcmc`, because the number of causals goes beyond 3 in the later case (I guess).

What is the best resource to get an idea about parametrization of the sampling procedure? Any particular paper of PAINTOR?

Thanks again for your answers.

**Comment by variani** [comment]
My other 5 cents on why it would be handy to control on `--n-causal-snps`. Following the recent review from FINEMAP authors (PMID [28942963](https://www.ncbi.nlm.nih.gov/pubmed/28942963)), misspecification of LD is a major issue in real data analysis of especially large-scale datasets. When "the posterior probability of the number of causal variants concentrates on the maximum value possible", it is an indication of the LD mismatch problem. Such a sanity check was very useful in my real data analysis to cope with inconsistent fine-mapping results.

I guess you can close the ticket. Thanks.
